[ao] correctly set public v private for fake_quantize.py

HDCharles opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #86037
* #86036
* #86034
* #86033
* #86032
* #86031
* #86030
* #86029
* #86028
* #86027
* #86026
* #86025
* #86024
* #86023
* __->__ #86022

Summary: biggest issue was that the constructors for the fake_quantize
classes use custom partials that live in the observer module and so
the module for these needed to be set correctly in the constructor class
method

Test Plan: python test/test_public_bindings.py

Reviewers:

Subscribers:

Tasks:

Tags:

HDCharles committed:
[ao] correctly set public v private for fake_quantize.py

Summary: biggest issue was that the constructors for the fake_quantize
classes use custom partials that live in the observer module and so
the module for these needed to be set correctly in the constructor class
method

Test Plan: python test/test_public_bindings.py

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 commented on torch/ao/quantization/fake_quantize.py:
```diff
@@ -18,6 +18,33 @@
 from typing import Any, Tuple
 
+__all__ = [
+    "FakeQuantizeBase",
```
do we want to fix what is visible to users as well? i.e. hide the things that we don't think makes sense to be public?

HDCharles replied:
is that the intent of this? I thought it was more about what things will be accessed by other modules

jerryzh168 replied:
I remember it's both to clean up the json file and also make the correct public/private decision for our apis, but we can confirm in team meeting

anjali411 replied:
Everything in `__all__` is supposed to be public and documented. If it's not meant to be public, then it should be preceded with an underscore.

HDCharles committed:
Update on "[ao] correctly set public v private for fake_quantize.py"


Summary: biggest issue was that the constructors for the fake_quantize
classes use custom partials that live in the observer module and so
the module for these needed to be set correctly in the constructor class
method

Test Plan: python test/test_public_bindings.py

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

HDCharles committed:
Update on "[ao] correctly set public v private for fake_quantize.py"


Summary: biggest issue was that the constructors for the fake_quantize
classes use custom partials that live in the observer module and so
the module for these needed to be set correctly in the constructor class
method

Test Plan: python test/test_public_bindings.py

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

HDCharles commented:
@pytorchmergebot merge
